src/services/auth.ts

```
@@ -49,11 +80,29 @@ export class AuthService {
-      if (!hasPermission) {
-        throw new ForbiddenError(KnownErrorCode.INSUFFICIENT_PERMISSIONS);
+
+      if (options?.permission_checker) {
+        let has_permission: boolean;
+
+        // Simple role-based check
+        if (Array.isArray(options.permission_checker)) {
+          has_permission = options.permission_checker.some((role) => user_roles.includes(role));
+        } else {
+          // Dynamic permission check
+          const staff_permissions = await this.getUserStaffPermissions(user.id, user_roles);
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)